homeassistant/components/honeygain/sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Sensors for HoneyGain data."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_DOLLAR
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HoneygainData
+from .const import DOMAIN
+
+
+@dataclass
+class SensorValueEntityDescription(SensorEntityDescription):
+    """Class describing Honeygain sensor entities."""
+
+    value: Callable = lambda x: x
+
+
+HONEYGAIN_SENSORS: list[SensorValueEntityDescription] = [
+    SensorValueEntityDescription(
+        key="account_balance",
+        name="Account Balance",
+        icon="mdi:hand-coin",
```

Comment:
I was once told that sensors with a device_class should not set an icon. But maybe with the `today_earnings` it is more clear.

Reply:
Thanks, updated the sensor descriptions